client/allocrunner/taskrunner/task_runner_test.go

```diff
@@ -2021,3 +2021,51 @@ func testWaitForTaskToStart(t *testing.T, tr *TaskRunner) {
 		require.NoError(t, err)
 	})
 }
+
+// TestTaskRunner_BaseLabels tests that the base labels for the task metrics
+// are set appropriately.
+func TestTaskRunner_BaseLabels(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+
+	alloc := mock.BatchAlloc()
+	alloc.Namespace = "not-default"
+	task := alloc.Job.TaskGroups[0].Tasks[0]
+	task.Driver = "raw_exec"
+	task.Config = map[string]interface{}{
+		"command": "whoami",
+	}
+
+	config, cleanup := testTaskRunnerConfig(t, alloc, task.Name)
+
+	tr, err := NewTaskRunner(config)
+	require.NoError(err)
+	defer cleanup()
+
+	var foundJob, foundGroup, foundTask, foundAlloc, foundNamespace bool
+	for _, e := range tr.baseLabels {
+		switch e.Name {
```

Comment:
this looks reasonable already, but I may suggest converting baselabels to a map then assert on map keys, e.g.

```golang
labels := map[string]string{}
for _, e := range tr.baseLabels {
  labels[e.Name] = e.Value
}

require.Equal(alloc.Job.Name, labels["job"])
require.Equal(alloc.TaskGroup, labels["task_group"])
```

I find this a tad more declarative and readable but it's not a big deal.

Reply:
thanks.